From 025b3f96b3b45e8e83d6c595c3fdd3def94da330 Mon Sep 17 00:00:00 2001
From: =?utf-8?q?Marc-Andr=C3=A9=20Lureau?= <marcandre.lureau@gmail.com>
Date: Wed, 7 Jan 2009 18:18:26 +0200
Subject: [PATCH 45/87] build: fix build on scratchbox with rpath

X11 depends on libraries that are not listed during link. Those
libraries symbol fail to be found, because -rpath seems to override
path to look the library for. Adding rpath-link with location of those
X11 library solves it.. Geezz...
---
 src/Makefile.am |    2 +-
 1 files changed, 1 insertions(+), 1 deletions(-)

diff --git a/src/Makefile.am b/src/Makefile.am
index cfc7635..5e5ac8a 100644
--- a/src/Makefile.am
+++ b/src/Makefile.am
@@ -154,7 +154,7 @@ PREOPEN_LIBS = $(modlibexec_LTLIBRARIES)
 endif
 
 if FORCE_PREOPEN
-pulseaudio_LDFLAGS = $(AM_LDFLAGS) $(BINLDFLAGS) -dlpreopen force $(foreach f,$(PREOPEN_LIBS),-dlpreopen $(f)) -Wl,--rpath -Wl,./.libs
+pulseaudio_LDFLAGS = $(AM_LDFLAGS) $(BINLDFLAGS) -dlpreopen force $(foreach f,$(PREOPEN_LIBS),-dlpreopen $(f)) -Wl,--rpath -Wl,./.libs -Wl,-rpath-link -Wl,$(libdir)
 else
 pulseaudio_LDFLAGS = $(AM_LDFLAGS) $(BINLDFLAGS) -dlopen force $(foreach f,$(PREOPEN_LIBS),-dlopen $(f))
 endif
-- 
1.6.3.3

